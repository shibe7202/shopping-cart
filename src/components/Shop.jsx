import useItemURL from './useItemUrl';
import Card from './Card'

function Shop({ cartTools }) {

    const { itemURL, error, loading } = useItemURL()


    return (
        <div className="shop">
            <p className="loading">{loading ? 'Loading...' : null}</p>
            <p className="error">{error ? 'A network error has occurred' : null}</p>
            <div className="cardGrid">
                {itemURL ? itemURL.map(data =>
                    <Card data={data} cartTools={cartTools} key={data.id} />
                ) : null}

            </div>

        </div>
    )
}

export default Shop