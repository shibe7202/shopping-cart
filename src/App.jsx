import { useState } from 'react'
import useCart from './components/useCart.jsx'
import './App.css'
import Shop from './components/Shop.jsx'

function App() {
  const cartTools = useCart()

  return (
    <>
      <Shop cartTools={cartTools} />
    </>
  )
}

export default App
