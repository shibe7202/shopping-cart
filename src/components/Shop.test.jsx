import { vi, describe, it, expect, afterEach } from "vitest";
import { render, screen } from "@testing-library/react";
import Shop from "./Shop";
import useItemURL from "./useItemUrl";
import App from '../App.jsx'

vi.mock('./useItemUrl')

describe("Shop component", () => {
    afterEach(() => {
        vi.resetAllMocks()
    })

    it("renders shopping page", () => {
        vi.mocked(useItemURL).mockReturnValue({
            loading: false,
            error: false,
            itemURL: [{ id: 'Item1' }, { id: 'Item2' }]
        })
        const { container } = render(<Shop />);
        expect(container).toMatchSnapshot();
    });

    it("renders loading screen", () => {
        vi.mocked(useItemURL).mockReturnValue({
            loading: true,
            error: false,
            itemURL: [{ id: 'Item1' }, { id: 'Item2' }]
        })

        render(<Shop />)
        expect(screen.getByText('Loading...').textContent).toMatch(/Loading.../)
    });

    it("renders error screen", () => {
        vi.mocked(useItemURL).mockReturnValue({
            loading: false,
            error: true,
            itemURL: [{ id: 'Item1' }, { id: 'Item2' }]
        })

        render(<Shop />)
        expect(screen.getByText('A network error has occurred').textContent).toMatch(/A network error has occurred/)
    })

    it("renders grid of shopping items", () => {
        vi.mocked(useItemURL).mockReturnValue({
            loading: false,
            error: false,
            itemURL: [{ id: 'Item1' }, { id: 'Item2' }]
        })

        render(<Shop />)
        expect(screen.getByText('Item1').textContent).toMatch(/Item1/)
    })
});