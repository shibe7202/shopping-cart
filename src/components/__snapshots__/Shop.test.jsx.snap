// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Shop component > renders shopping items 1`] = `
<div>
  <div
    class="shop"
  >
    <p
      class="loading"
    />
    <p
      class="error"
    />
    <div
      class="cardGrid"
    >
      <p>
        Item1
      </p>
      <p>
        Item2
      </p>
    </div>
  </div>
</div>
`;

exports[`Shop component > renders shopping page 1`] = `
<div>
  <div
    class="shop"
  >
    <p
      class="loading"
    />
    <p
      class="error"
    />
    <div
      class="cardGrid"
    >
      <p>
        Item1
      </p>
      <p>
        Item2
      </p>
    </div>
  </div>
</div>
`;
